rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -180,6 +186,16 @@ void TimeSource::create_clock_sub()
     allocator);
 }
 
+void TimeSource::destroy_clock_sub()
+{
+  std::lock_guard<std::mutex> guard(clock_sub_lock_);
+  if (!clock_subscription_) {
+    // Subscription already destroyed/never created.
+    return;
+  }
+  clock_subscription_.reset();
```

Comment:
This call is always safe, so you can do it ever if `!clock_subscription_`.

Reply:
Updated in e34e0d71dc91143b9915aaa12dc247d4f79c59c9